packages/@azure/servicebus/data-plane/lib/core/messageReceiver.ts

```diff
@@ -1003,7 +1003,7 @@ export class MessageReceiver extends LinkEntity {
     }
     const rcvrOptions: ReceiverOptions = {
       name: useNewName ? getUniqueName(this._context.entityPath) : this.name,
-      autoaccept: false,
+      autoaccept: this.receiveMode === ReceiveMode.receiveAndDelete ? true : false,
```

Comment:
Recommend adding some tests for this PR if possible.

Reply:
Nevermind this, the performance tests would cover it and we don't really have a specific place for tests on the message receiver currently.